Makefile.tools

```diff
@@ -0,0 +1,289 @@
+# This Makefile fragment is included from Makefile and is not intended to be
+# used directly.
+
+# Convention: OBJ_TOOLS_BIN_<x> ↔ CMD_<x>
+
+# The lines beginning with "# _plsi_ are metadata by PLSI internally.
+# Lines like # _plsi_ contrib python targets = OBJ_TOOLS_BIN_PYTHON3 are used by build-contrib-auto
+# to automatically determine the binary targets to build and export.
+
+# Note: while having to manually add PKG_CONFIG_PATH is annoying, Make
+# lacks a way to easily perform conditional includes, and always including
+# would implictly create a dependency for all runs, even contrib builds.
+
+ifeq ($(OBJ_TOOLS_BIN_DIR),)
+$(error Set OBJ_TOOLS_BIN_DIR to the folder where binaries will be installed)
+endif
+
+ifeq ($(OBJ_TOOLS_SRC_DIR),)
+$(error Set OBJ_TOOLS_SRC_DIR to the folder where source files will be installed)
+endif
+
+ifeq ($(PLSI_CACHE_DIR),)
+$(error Set PLSI_CACHE_DIR to the PLSI cache dir)
+endif
+
+ifeq ($(SCHEDULER_CMD),)
+$(error Set SCHEDULER_CMD from Makefile)
+endif
+
+# Many of the Makefiles (e.g. in python3) rely in implicit rules to function,
+# so cancel out any -rR for them.
+TOOLS_MAKEFLAGS = $(shell echo $(MAKEFLAGS) | sed -e "s/r//g" | sed -e "s/R//g")
```

Comment:
Interesting I thought this was an environment variable that make would look at for flags. My mistake.